### PR TITLE
6351 packages sync

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|#6351|

## **Description**

This issue aims to make use of the generic synchronization libraries, to use the synchronization solution algorithm that both shared modules provide.

These changes have to be made in the various implementations according to the operating system targeted for this feature.

## **DoD**
- [X] Implementation
- [X] **Run testtool on Win**

- [X] **Run testtool on Linux**
image
- [X] **RTR syscollector**
image
- [X] **RTR dbsync**
image

### CODE REVIEW

**Review by tinachoB — APPROVED** [review]
LGTM!

File: src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp [hunk 1]
```diff
@@ -170,29 +169,37 @@ void SQLiteDBEngine::syncTableRowData(const std::string& table,
...
+                const bool diffExist { getRowDiff(primaryKeyList, table, entry, jsResult) };
+                if (diffExist)
                 {
-                    const auto& transaction { m_sqliteFactory->createTransaction(m_sqliteConnection)};
-                    updateSingleRow(table, jsDataToUpdate[0]);
-                    transaction->commit();
+                    const nlohmann::json jsDataToUpdate{getDataToUpdate(primaryKeyList, jsResult, entry, inTransaction)};
+                    if (!jsDataToUpdate[0].empty())
+                    {
+                        const auto& transaction { m_sqliteFactory->createTransaction(m_sqliteConnection)};
```
Comment (Dwordcito) [inline src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp:182]: this transaction is necessary ?

File: src/wazuh_modules/syscollector/syscollectorImp.cpp [hunk 2]
```diff
@@ -64,8 +75,10 @@ Syscollector::Syscollector(const std::shared_ptr<ISysInfo>& spInfo,
 , m_processes{processes}
 , m_hotfixes{hotfixes}
 , m_running{false}
+, m_dbSync{HostType::AGENT, DbEngineType::SQLITE3, "temp.db", getCreateStatement()}
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/syscollectorImp.cpp:78]: rename to syscollector.db

File: src/wazuh_modules/syscollector/syscollectorImp.cpp [hunk 3]
```diff
@@ -64,8 +75,10 @@ Syscollector::Syscollector(const std::shared_ptr<ISysInfo>& spInfo,
...
 , m_thread{std::bind(&Syscollector::syncThread, this)}
 {
+
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/syscollectorImp.cpp:81]: remove space, \n

File: src/wazuh_modules/syscollector/syscollectorImp.cpp [hunk 4]
```diff
@@ -105,13 +118,34 @@ void Syscollector::scanNetwork()
...
+            //notify changes
+            std::cout << result << std::endl;
+            std::cout << data.dump() << std::endl;
         }
+    };
+    DBSyncTxn txn
+    {
+        m_dbSync.handle(),
+        tables.at("tables"),
+        1,
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/syscollectorImp.cpp:137]: Set in 0, to get the harware_concurrency() result.

File: src/wazuh_modules/syscollector/syscollectorImp.cpp [hunk 5]
```diff
@@ -105,13 +118,34 @@ void Syscollector::scanNetwork()
...
+        1024,
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/syscollectorImp.cpp:138]: constexpr

**Review by Dwordcito — CHANGES_REQUESTED** [review]
minor changes!

File: src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp (hunk 1 above)
Comment (konangc87) [reply src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp:182]: for the statement bind within updateSingleRow

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**